### ISSUE TITLE
Problem: GUI might be fused to headless wallet

### ISSUE THREAD

**Issue by sjmackenzie** [fractalide/cantor#1]
Solution: 

The cardano wallet should be developed such that any dapp within the fractalide eco-system should be able to call a headless cardano wallet and be able to use the functionality.

So the hierarchy will look like two stacked triangles
![stacked-wallet](https://user-images.githubusercontent.com/158321/40356125-c3902f18-5dea-11e8-9f7e-91b20135ba28.png)

**Comment by clacke** [comment]
Yes, that's my aim. The wallet UI and the SL adapter should just be two nodes with reasonable APIs that we connect together, no hanky panky.
